sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -757,4 +764,75 @@
             return monoError(logger, ex);
         }
     }
+
+    /**
+     * Initiates a search and returns a {@link PhoneNumberSearch} usable by other functions
+     * This function returns a Long Running Operation poller that allows you to 
+     * wait indefinitely until the operation is complete.
+     * 
+     * @param options A {@link CreateSearchOptions} with the search options
+     * @param pollInterval The time our long running operation will keep on polling 
+     * until it gets a result from the server
+     * @return A {@link PollerFlux} object with the search result
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PollerFlux<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+        CreateSearchOptions options, Duration pollInterval) {
+        Objects.requireNonNull(options, "'options' cannot be null.");
+        Objects.requireNonNull(pollInterval, "'pollInterval' cannot be null.");
+        return new PollerFlux<PhoneNumberSearch, PhoneNumberSearch>(pollInterval,
+            createSearchActivationOperation(options),
+            createSearchPollOperation(),
+            cancelSearchOperation(),
+            createSearchFetchResultOperation());
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, Mono<PhoneNumberSearch>> 
+        createSearchActivationOperation(CreateSearchOptions options) {
+        return (pollingContext) -> {
+            Mono<PhoneNumberSearch> response = createSearch(options).flatMap(createSearchResponse -> {
+                String searchId = createSearchResponse.getSearchId();
+                Mono<PhoneNumberSearch> phoneNumberSearch = getSearchById(searchId);
+                return phoneNumberSearch;
+            });
+            return response;
+        };
+    }
+
+    private Function<PollingContext<PhoneNumberSearch>, 
+        Mono<PollResponse<PhoneNumberSearch>>> createSearchPollOperation() {
+        return pollingContext ->
+            getSearchById(pollingContext.getLatestResponse().getValue().getSearchId())
+                .flatMap(getSearchResponse -> {
+                    SearchStatus status = getSearchResponse.getStatus();
+                    if (status.equals(SearchStatus.SUCCESS) || 
```

Comment:
I don't think we should get SUCCESS in here, because that is after purchaseSearch.. but I don;t kow what is the right thing to do in that case..